src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -339,6 +339,28 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
             operands.Free();
         }
 
+        public override void VisitBinaryPattern(BinaryPatternSyntax node)
+        {
+            // Like binary expressions, binary patterns are left-associative, and can be deeply nested (even in our own code).
+            // Handle this with manual recursion.
+            PatternSyntax currentPattern = node;
+
+            var rightPatternStack = ArrayBuilder<PatternSyntax>.GetInstance();
+
+            while (currentPattern is BinaryPatternSyntax binaryPattern)
```

Comment:
>while (currentPattern is BinaryPatternSyntax binaryPattern) [](http://example.com/codeflow?start=12&length=59)

Consider using a loop with a post-condition. We know the condition is true on entry.

Reply:
I think that results in less-clear code, like we have in `VisitBinaryExpression` above. I'd prefer to keep this as is.